internal/controller/supervisorconfig/githubupstreamwatcher/github_upstream_watcher.go

```diff
@@ -93,52 +100,210 @@ func (c *gitHubWatcherController) Sync(ctx controllerlib.Context) error {
 		return fmt.Errorf("failed to list GitHubIdentityProviders: %w", err)
 	}
 
-	var errs []error
+	var applicationErrors []error
 
-	requeue := false
 	validatedUpstreams := make([]upstreamprovider.UpstreamGithubIdentityProviderI, 0, len(actualUpstreams))
 	for _, upstream := range actualUpstreams {
-		valid, err := c.validateUpstream(ctx, upstream)
-		if valid == nil {
-			requeue = true
-			errs = append(errs, err)
-		} else {
-			validatedUpstreams = append(validatedUpstreams, upstreamprovider.UpstreamGithubIdentityProviderI(valid))
+		validatedUpstream, applicationErr := c.validateUpstreamAndUpdateConditions(ctx, upstream)
+		if applicationErr != nil {
+			applicationErrors = append(applicationErrors, applicationErr)
+		} else if validatedUpstream != nil {
+			validatedUpstreams = append(validatedUpstreams, validatedUpstream)
 		}
+		// Else:
+		// If both validatedUpstream and applicationErr are nil, this must be because the upstream had configuration errors.
```

Comment:
👍 this is the "there is a status that is an "error" but there is no returned err from the Sync" case yeah?

Reply:
Yes.